openedx/features/course_experience/utils.py

```diff
@@ -275,3 +280,29 @@ def reset_deadlines_banner_should_display(course_key, request):
                         display_reset_dates_banner = True
                         break
     return display_reset_dates_banner
+
+
+def can_show_verified_upgrade(user, course_id, enrollment):
+    """
+    Check if we are able to show verified upgrade message based
+    on the enrollment and current user partition
+    """
+    if not enrollment:
+        return False
+    partition_service = PartitionService(course_id)
```

Comment:
Could you explain the significance of user partitions or link to me some documentation? I haven't been able to find information about what they're used for. Are user partitions what are used to implement masquerading? Or are they something else?

Reply:
See https://openedx.atlassian.net/wiki/spaces/MS/pages/18350129/Group-Based+Access+Control